### DISCUSSION TITLE
Help with server stuck after "compiled successfully" with "10% building 0/1 entries 0/0 dependencies 0/0 modules"

### DISCUSSION THREAD

**Discussion by MagiusCHE** [webpack/webpack#16216] [Q&A]
Hi guys!
My problem is this: i just develop my app on 2 machines.
1. Linux Manjaro (and all works)
2. Windows 11 (and here the problem)

When i just start `webpack --config webpack.config.js` it compiles successfully. but when i run `webpack serve --progress` after successfully recompilation another "10% building..." appears and all become stucked with NodeJS process overloaded in background.

On these conditions i can't even browse project. Every call at `http://localhost:8080` stucks in pending forever.

![Screenshot_20220904_185043](https://user-images.githubusercontent.com/46496052/188325031-6f6cf68a-b9b9-47db-ad52-f5ba4216a307.png)

webpack.config.js
```javascript
const CopyWebpackPlugin = require("copy-webpack-plugin");
const path = require('path');

//const ASSET_PATH = process.env.ASSET_PATH || './';

module.exports = {
  entry: "./bootstrap.js",
  devtool: 'inline-source-map',
  parallelism: 1,
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: /node_modules/,
      },
      {
        test: /\.css$/,
        use: ['style-loader', 'css-loader']
      },
      {
        test: /\.(scss)$/,
        use: [{
          loader: 'style-loader', // inject CSS to page
        }, {
          loader: 'css-loader', // translates CSS into CommonJS modules
        }, {
          loader: 'postcss-loader', // Run post css actions
          options: {
            postcssOptions: {
              plugins: function() { // post css plugins, can be exported to postcss.config.js
                return [
                  require('precss'),
                  require('autoprefixer')
                ];
              }
            }
          }
        }, {
          loader: 'sass-loader' // compiles Sass to CSS
        }]
      },
      {
        test: /\.(woff|woff2)$/,
        use: {
          loader: 'url-loader',
        },
      },
    ],
  },
  resolve: {
    extensions: ['.tsx', '.ts', '.js'],
  },
  output: {
    path: path.resolve(__dirname, "../../www"),
    filename: "bootstrap.js",
    //publicPath: ASSET_PATH,
  },
  mode: "development",
  plugins: [
    new CopyWebpackPlugin({ patterns: ['index.html', { from: 'assets//', to: 'assets//' }] })
  ],
  experiments: {
    asyncWebAssembly: true,
  },
  devServer: {
    hot: false,
    liveReload: false,
    client: {
      overlay: false,
    },
  },
};
```
Can someone help me to figure out the problem?

If you need other info just ask!

Thx in advance!

**Comment by alexander-akait** [comment]
```
new CopyWebpackPlugin({ patterns: ['index.html', { from: 'assets//', to: 'assets//' }] })
```

looks like ininity loop, can you try to disable it and check it

**Reply by MagiusCHE** [reply]
> ```
> new CopyWebpackPlugin({ patterns: ['index.html', { from: 'assets//', to: 'assets//' }] })
> ```
> 
> looks like ininity loop, can you try to disable it and check it

Nope... nothing change (but thanks for the advice).
![immagine](https://user-images.githubusercontent.com/46496052/188497203-3bd4a6ba-908d-4387-b527-1f85c17055cc.png)

**Reply by alexander-akait** [reply]
Do you have any non official plugins? Can you try another shell (maybe just bug in shell in your os)?

**Comment by MagiusCHE** [comment]
Gotha!!!!!!

It is PowerShell!!!!!!

If i run cmd.exe all works correctly.

Thx for all!

**Comment by MagiusCHE** [comment]
Damn me...
sorry but it was a false "success" :-(
I dont know how it worked one time but... now the problem is the same...
Just to be sure i do this:
- Open `cmd.exe`
- run `webpack serve --progress`
- got `Node.js Javascript Runtime` consumes a lot of ram and processor.
![immagine](https://user-images.githubusercontent.com/46496052/188514788-084f6fc1-188a-4f5b-9f92-e8096410ea24.png)
- cant connect using browser via `http://localhost:8080` (the browser remains in pending)
![immagine](https://user-images.githubusercontent.com/46496052/188514828-87952990-b0b6-412b-9231-45720dd126e7.png)

So now i'm stucked again.... (in the meantime i'll try to remove node_modules again end rebuild all via cmd.exe)
---- EDIT ----
btw, node_modules removed and yarn install-ed all..without success.


Thx in advance.

**Reply by alexander-akait** [reply]
Do you try to bundle something big?

**Reply by MagiusCHE** [reply]
> Do you try to bundle something big?

Let me clear....
- I build the package without problem (`"webpack --config webpack.config.js --progress`) (under linux and winndows 11)
- I can even build the final "executable" (for linux and window).
- I can run the final executable without issues (under linux and windows)
- I can run `webpack serve --progress` under linux and browse localhost without problem.

The only problem i have is running `webpack serve --progress` under windows 11.... it just hangs all...
Dist dir is:
![immagine](https://user-images.githubusercontent.com/46496052/188516775-95e4a71c-ffb7-458f-a040-82eab9b57a62.png)

**Reply by alexander-akait** [reply]
> The only problem i have is running webpack serve --progress under windows 11.... it just hangs all...

Sounds like a firewall or antivirus

**Reply by MagiusCHE** [reply]
> Sounds like a firewall or antivirus

I have no antivirus, plus i've disabled firewall entirly..... :-( but nothing...
The strange thing is that after a `webpack serve` the NodeJS process continues consuming resource in background... basically forever...at 10-20% of the processor and increasing memory allocation....
![immagine](https://user-images.githubusercontent.com/46496052/188555359-55e9949a-fac1-4c7c-ab77-4dfff7950b79.png)

**Reply by alexander-akait** [reply]
Do you have the `public` directory? Maybe chokidar are watching more and more files...

**Reply by MagiusCHE** [reply]
no... that directory not exists. (actually i dont event know how to prevent webpack to watch in there... i just not need "public static non-webpack" contents).

**Reply by alexander-akait** [reply]
Is here https://github.com/webpack/webpack/discussions/16216#discussion-4360690 your full configuration?

**Reply by MagiusCHE** [reply]
Yes, actually i've modified it into this one. With same results:
```javascript
const CopyWebpackPlugin = require("copy-webpack-plugin");
const path = require('path');

//const ASSET_PATH = process.env.ASSET_PATH || './';

module.exports = {
  entry: "./bootstrap.js",
  devtool: 'inline-source-map',

  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: /node_modules/,
      },
      {
        test: /\.css$/,
        use: ['style-loader', 'css-loader']
      },
      {
        test: /\.(scss)$/,
        use: [{
          loader: 'style-loader', // inject CSS to page
        }, {
          loader: 'css-loader', // translates CSS into CommonJS modules
        }, {
          loader: 'postcss-loader', // Run post css actions
          options: {
            postcssOptions: {
              plugins: function() { // post css plugins, can be exported to postcss.config.js
                return [
                  require('precss'),
                  require('autoprefixer')
                ];
              }
            }
          }
        }, {
          loader: 'sass-loader' // compiles Sass to CSS
        }]
      },
      {
        test: /\.(woff|woff2)$/,
        use: {
          loader: 'url-loader',
        },
      },
    ],
  },
  resolve: {
    extensions: ['.tsx', '.ts', '.js'],
  },
  optimization: {
    splitChunks: {
      cacheGroups: {
        defaultVendors: {
          filename: 'vendor.bundle.js',
        },
      },
    },
  },
  output: {
    path: path.resolve(__dirname, "../../www"),
    filename: "[name].bootstrap.js",
    clean: true,
    publicPath: '/'
  },
  optimization: {
    runtimeChunk: 'single',
  },
  mode: "development",
  plugins: [
    new CopyWebpackPlugin({ patterns: ['index.html', { from: 'assets//', to: 'assets//' }] })
  ],
  experiments: {
    asyncWebAssembly: true,
  },

  watchOptions: {
    ignored: './src/www/public/.*',
  },
  devServer: {
    hot: false,
    liveReload: false,
    static: {
      directory: path.resolve(__dirname, "../../www"),
      watch: false,
    },
    client: {
      overlay: false,
    },
  },
};

```

**Reply by alexander-akait** [reply]
Hm, you have `asyncWebAssembly`, do you have something with WebAssembly? Also `url-loader` is deprecated in favor https://webpack.js.org/guides/asset-modules/

**Reply by MagiusCHE** [reply]
Yes i have `WebAssembly`. However i just commented out  `url-loader`
![immagine](https://user-images.githubusercontent.com/46496052/188965193-ead9e5d4-dd1c-499d-bfd1-2cc9c9c4359a.png)
but nothing changes.

Consider this. I just run `webpack serve --progress` and after that (without even open browser) the `NodeJS` process start drain processor and ram. 
![immagine](https://user-images.githubusercontent.com/46496052/188965816-448cfbb4-9354-4cd4-a9a4-7b89236ab834.png)
![immagine](https://user-images.githubusercontent.com/46496052/188965946-b22f5d5e-0c2a-459b-9ee0-678364ec05a6.png)
And if i try browse it, all requests hangs in pendings...
So the problem stays just in `webpack serve` i suppose....

**Reply by alexander-akait** [reply]
Can you check your reqest reached this line https://github.com/webpack/webpack-dev-middleware/blob/master/src/middleware.js#L62? Also check request in dev server too. Unfortunately, I can not advise anything else, I recommend trying to debug

**Reply by MagiusCHE** [reply]
yha... i guess my only chance is try debugging internal webpack code.
Just to be clear, (my english is very poor sorry) but i figure out the problem is not in Requesting process to webserver, it is inside the "serve" procress prior to any requests.
Infact even if i just call `webpack serve` the cpu and mem starts to grow without ending.
Making a request to that "busy" server, actually stay in pending and in rare case single response may be received but... due to that heavy busy state the express server can't do much more.

So i just try to focus debugging "serve" process... but i guess i can't be so skilled to figure much.

Btw thx for your help and time, i try to put news here eventually (if i found something).

**Reply by alexander-akait** [reply]
Yeah, feel free to provide more information and ping me, but it really sounds like antivirus (or other background software which snan memory/files), can you try it on another window computer (with the same version of window)
